File: Implementations/PrimeCheck.kt
import kotlin.math.sqrt
import kotlin.system.exitProcess
fun main() 
{
    println("Enter a number")
    val num = readLine()!!.toULong()
    for (i in 2uL..sqrt(num.toDouble()).toULong())
    {
        if (num % i == 0uL)
        {
            println("$num is not prime: $i*${num / i}")
            exitProcess(0)
        }
    }
    println("$num is prime")
}

File: Implementations/PrimeCheck.zig
const std = @import("std");

pub fn main() !void 
{
    const stdout = std.io.getStdOut().writer();
    const stdin = std.io.getStdIn().reader();
    var buff: [50]u8 = undefined;
    var fba = std.heap.FixedBufferAllocator.init(&buff);
    const alloc = fba.allocator();
    
    try stdout.print("Enter a number\n", .{});

    const input = try stdin.readUntilDelimiterOrEofAlloc(alloc, '\n', 100);
    const a = input.?;
    const num = try std.fmt.parseInt(u128, a, 10);
    
    for (2..std.math.sqrt(num) + 1) |i|
    {
        if (num % i == 0)
        {
            try stdout.print("{} is not prime: {}*{}\n", .{num, i, num / i});
            return;
        }
    }

    try stdout.print("{} is prime\n", .{num});
    return;
}
